src/Components/Web.JS/src/Rendering/DomMerging/DataPermanentElementSync.ts

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+export function isDataPermanentElement(elem: Element): boolean {
+  return elem.hasAttribute('data-permanent');
+}
+
+export function areIncompatibleDataPermanentElements(elementA: Element, elementB: Element) {
```

Comment:
```suggestion
export function cannotMergeDueToDataPermanentAttributes(elementA: Element, elementB: Element): boolean {
```

... because the current name implies that, if it returns `true`, the elements are "data permanent elements".